dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java

```diff
@@ -114,7 +114,7 @@ public RowServiceInputStream(DataPartition dp, FieldDef rd, FieldDef pRd, int co
         this.readBlock();
     }
 
-    private boolean setNextFilePartCopy()
+    private synchronized boolean setNextFilePartCopy()
```

Comment:
Any reason to make this synchronized? Will multiple threads hit this function? I don't see a problem with marking in synchronized, but wanted to make sure I wasn't missing something.

Reply:
Yes, the tardis.hpcc package uses rowServiceInputStream in a multithreaded fashion.

Reply:
@drealeed RowServiceInputStream itself isn't thread safe. So each thread should be either be using their own RowServiceInputStream or handling synchronization at a higher level. RowServiceInputStream should be viewed as a typical Java InputStream with no thread safety guarantees

Reply:
ok agreed, the synchronization at this level is overkill, and might set incorrect expectation

Reply:
@jpmcmu good to know, I'll review the hpcc tardis dfu reader and make whatever changes are needed.